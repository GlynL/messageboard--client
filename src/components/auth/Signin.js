import React from "react";
import { connect } from "react-redux";
import * as actions from "../../actions/auth";

import Form from "./Form";

export const Signin = props => {
  function handleSubmit(user) {
    props.signin(user, () => props.history.push("/"));
  }

  return (
    <div className="container">
      <h1>Sign In</h1>
      {props.errorMessage && <div>{props.errorMessage}</div>}
      <Form handleSubmit={handleSubmit} buttonText="Sign In" />
    </div>
  );
};

const mapStateToProps = state => ({ errorMessage: state.auth.errorMessage });

export default connect(
  mapStateToProps,
  actions
)(Signin);
