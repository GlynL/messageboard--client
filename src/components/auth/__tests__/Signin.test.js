import React from "react";
import { shallow } from "enzyme";
import { Signin } from "../Signin";

describe("w/out props", () => {
  let wrapper, history, signin;
  beforeEach(() => {
    history = { push: jest.fn() };
    signin = jest.fn();
    wrapper = shallow(<Signin history={history} signin={signin} />);
  });

  it("renders Signin correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should handleChange of email input", () => {
    wrapper
      .find("input[name='email']")
      .simulate("change", { target: { name: "email", value: "test email" } });
    expect(wrapper.state("email")).toBe("test email");
  });

  it("should handleChange of password input", () => {
    wrapper.find('input[name="password"]').simulate("change", {
      target: { name: "password", value: "test password" }
    });
    expect(wrapper.state("password")).toBe("test password");
  });

  it("should handleSubmit of form", () => {
    const preventDefault = jest.fn();
    wrapper.setState({ email: "test email", password: "test password" });
    wrapper.find("form").simulate("submit", { preventDefault });
    expect(preventDefault).toHaveBeenCalled();
    expect(signin).toHaveBeenCalled();
    expect(wrapper.state("email")).toBe("");
    expect(wrapper.state("password")).toBe("");
  });
});

it("renders Signin with error prop correctly", () => {
  const wrapper = shallow(<Signin errorMessage="test error" />);
  expect(wrapper).toMatchSnapshot();
});
