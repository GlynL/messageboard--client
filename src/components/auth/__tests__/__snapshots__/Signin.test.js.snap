// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Signin with error prop correctly 1`] = `
<div>
  <h1>
    Sign In
  </h1>
  <div>
    test error
  </div>
  <Form
    buttonText="Sign In"
    handleSubmit={[Function]}
  />
</div>
`;

exports[`w/out props renders Signin correctly 1`] = `
<div>
  <h1>
    Sign In
  </h1>
  <Form
    buttonText="Sign In"
    handleSubmit={[Function]}
  />
</div>
`;
